File: repos_estudiantes_TARQUI_segunda_entrega/Mallcu_Calani_Alejandro_Gabriel/Composicion Agregacion/Ejercicio3/Ejercicio3.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso

    def mostrar_info(self):
        return f"- {self.nombre}: {self.peso} kg"

    def get_nombre(self):
        return self.nombre

    def get_peso(self):
        return self.peso


class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []

    def agregar_parte(self, parte):
        self.partes.append(parte)

    def mostrar_avion(self):
        print(f"Modelo: {self.modelo}")
        print(f"Fabricante: {self.fabricante}")
        print("Partes:")
        for p in self.partes:
            print(p.mostrar_info())


# b) Crear avión y agregar partes
avion = Avion("Boeing 747", "Boeing")
avion.agregar_parte(Parte("Motor", 3000))
avion.agregar_parte(Parte("Ala Izquierda", 1500))
avion.agregar_parte(Parte("Ala Derecha", 1500))
avion.agregar_parte(Parte("Tren de Aterrizaje", 1200))

# c) Mostrar información
avion.mostrar_avion()

File: repos_estudiantes_ALVAREZ_segunda_entrega/Alba_Plata_Bryan_Alberto/PracticaDOS/ejercicioCom3/Ejercicio3/src/ejercicio3/Ejercicio3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package ejercicio3;

/**
 *
 * @author Usuario
 */
import java.util.*;

class Parte {
    private String nombre;
    private double peso;

    public Parte(String nombre, double peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public void mostrar_info() {
        System.out.println("Parte: " + nombre + ", Peso: " + peso + " kg");
    }
}

class Avion {
    private String modelo;
    private String fabricante;
    private List<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        partes = new ArrayList<>();
    }

    public void agregar_parte(Parte p) {
        partes.add(p);
    }

    public void mostrar_avion() {
        System.out.println("Avión Modelo: " + modelo + ", Fabricante: " + fabricante);
        for (Parte p : partes) {
            p.mostrar_info();
        }
    }
}
public class Ejercicio3 {
    public static void main(String[] args) {
        Avion avion = new Avion("Boeing 737", "Boeing");
        avion.agregar_parte(new Parte("Motor", 1500));
        avion.agregar_parte(new Parte("Ala izquierda", 500));
        avion.agregar_parte(new Parte("Tren de aterrizaje", 800));
        avion.mostrar_avion();
    }
    
}
